platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/SimpleTestEventLogger.java

```diff
@@ -72,13 +72,13 @@ public void completed(TestDescriptorInternal descriptor, TestResult result, Test
                 for (TestFailure failure : result.getFailures()) {
                     TestFailureDetails details = failure.getDetails();
                     if (!TextUtil.isBlank(details.getMessage())) {
-                        if (details.isAssertionFailure()) {
-                            // test assertion (should be most common)
-                            output.append("    ").withStyle(StyledTextOutput.Style.Failure).println(details.getMessage());
-                        } else if (details.isFileComparisonFailure()) {
+                        if (details.isFileComparisonFailure()) {
                             // comparison failure
                             output.append("    Expected: ").withStyle(StyledTextOutput.Style.Failure).println(details.getExpected());
                             output.append("    Actual: ").withStyle(StyledTextOutput.Style.Success).println(details.getActual());
+                        } else if (details.isAssertionFailure()) {
+                            // test assertion (should be most common)
```

Comment:
Why switch the order here? 

The comment states that a test assertion should be the most common scenario, so it seems to make sense that we check it first

Though if we are no longer checking it first, then the `(should be most common)` comment doesn't really make sense anymore imo

Reply:
I didn't realize it when this was first written, but there was no way to create a file comparison failure that wasn't also an assertion failure.  The tests for this created these manually, but that's not how they're created through DefaultTestFailure